zkevm-circuits/src/evm_circuit/execution/sload.rs

```diff
@@ -0,0 +1,409 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        step::ExecutionState,
+        table::CallContextFieldTag,
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{
+                ConstraintBuilder, StepStateTransition,
+                Transition::{Delta, To},
+            },
+            select, Cell, Word,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+use eth_types::{evm_types::GasCost, Field, ToLittleEndian, ToScalar};
+use halo2_proofs::{
+    circuit::Region,
+    plonk::{Error, Expression},
+};
+
+#[derive(Clone, Debug)]
+pub(crate) struct SloadGadget<F> {
+    same_context: SameContextGadget<F>,
+    tx_id: Cell<F>,
+    rw_counter_end_of_reversion: Cell<F>,
+    is_persistent: Cell<F>,
+    callee_address: Cell<F>,
+    key: Word<F>,
+    value: Word<F>,
+    committed_value: Word<F>,
```

Comment:
For these three values you're using `Word`, which allocates 32 cells for each, and when you call `expr()` you get the RLC encoding of the word.
I see that in the gadget these values are only used as RLC encoded (their bytes/limbs are never used).  So maybe there's no need to use `Word` here and you can leave them as `Cell`?  For example, check the implementation of `CALLVALUE`: https://github.com/appliedzkp/zkevm-circuits/blob/5254119c8cbecbe5aa6b6df84f427e15e05a5b08/zkevm-circuits/src/evm_circuit/execution/callvalue.rs#L24

Reply:
Sounds good, I will update it!